libr/anal/p/anal_xtensa.c

```diff
@@ -634,6 +634,12 @@ static inline void sign_extend(st32 *value, ut8 bit) {
 	}
 }
 
+static inline void sign_extend2(st32 *value, ut8 bit1, ut8 bit2, ut8 shift) {
+	if (((*value >> bit1) & 1) && ((*value >> bit2) & 1)) {
+		*value |= UT32_MAX << (32 - shift);
```

Comment:
Changed to 32, but 31 is correct too but 32 has more sense: for example if shift==25 than 32-25=7 equal to size of sign extented operand (i.e. 7 bit signed value extended to 32 bit signed one)